osbuild/buildroot.py

```diff
@@ -169,7 +169,7 @@ def register_api(self, api: "BaseAPI"):
         if self._exitstack:
             self._exitstack.enter_context(api)
 
-    def run(self, argv, monitor, timeout=None, binds=None, readonly_binds=None):
+    def run(self, argv, monitor, timeout=None, binds=None, readonly_binds=None, extra_env=None):
```

Comment:
I almost think we could make this a class-level property. I have always planned to re-use the the same build root instance for all invocations of stages of a single pipeline since all stage-dependent things are bind-mounts and the build root is shared among them.

Having said that, this is fine for now, I can always fix this up when I get to that optimisation.